File: mobile/src/main/java/treehou/se/habit/ui/bindings/BindingFragment.java
package treehou.se.habit.ui.bindings;

import android.os.Bundle;
import android.support.v4.app.Fragment;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import butterknife.BindView;
import butterknife.ButterKnife;
import butterknife.Unbinder;
import se.treehou.ng.ohcommunicator.util.GsonHelper;
import se.treehou.ng.ohcommunicator.connector.models.OHBinding;
import treehou.se.habit.R;

public class BindingFragment extends Fragment {

    private static final String ARG_BINDING = "ARG_BINDING";

    @BindView(R.id.lbl_name) TextView lblName;
    @BindView(R.id.lbl_author) TextView lblAuthor;
    @BindView(R.id.lbl_description) TextView lblDescription;

    private OHBinding binding;
    private Unbinder unbinder;

    public static BindingFragment newInstance(OHBinding binding) {
        BindingFragment fragment = new BindingFragment();
        Bundle args = new Bundle();
        args.putString(ARG_BINDING, GsonHelper.createGsonBuilder().toJson(binding));
        fragment.setArguments(args);
        return fragment;
    }

    public BindingFragment() {
        // Required empty public constructor
    }

    @Override
    public void onCreate(Bundle savedInstanceState) {

        if(getArguments() != null){
            if(getArguments().containsKey(ARG_BINDING)){
                binding = GsonHelper.createGsonBuilder().fromJson(getArguments().getString(ARG_BINDING), OHBinding.class);
            }
        }

        super.onCreate(savedInstanceState);
    }

    @Override
    public View onCreateView(LayoutInflater inflater, ViewGroup container,
                             Bundle savedInstanceState) {

        // Inflate the layout for this fragment
        View rootView = inflater.inflate(R.layout.fragment_binding, container, false);
        unbinder = ButterKnife.bind(this, rootView);

        lblName.setText(binding.getName());
        lblAuthor.setText(binding.getAuthor());
        lblDescription.setText(binding.getDescription());

        return rootView;
    }

    @Override
    public void onDestroyView() {
        super.onDestroyView();
    unbinder.unbind();
    }
}

File: mobile/src/main/java/treehou/se/habit/ui/bindings/BindingFragment.kt
package treehou.se.habit.ui.bindings

import android.os.Bundle
import android.support.v4.app.Fragment
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import com.crashlytics.android.Crashlytics
import kotlinx.android.synthetic.main.fragment_binding.*
import se.treehou.ng.ohcommunicator.connector.models.OHBinding
import se.treehou.ng.ohcommunicator.util.GsonHelper
import treehou.se.habit.R
import treehou.se.habit.util.Constants

class BindingFragment : Fragment() {

    private var binding: OHBinding? = null

    override fun onCreate(savedInstanceState: Bundle?) {

        Crashlytics.setString(Constants.FIREABASE_DEBUG_KEY_FRAGMENT, javaClass.name)
        if (arguments != null) {
            if (arguments!!.containsKey(ARG_BINDING)) {
                binding = GsonHelper.createGsonBuilder().fromJson(arguments!!.getString(ARG_BINDING), OHBinding::class.java)
            }
        }

        super.onCreate(savedInstanceState)
    }

    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?,
                              savedInstanceState: Bundle?): View? {

        // Inflate the layout for this fragment
        return inflater.inflate(R.layout.fragment_binding, container, false)
    }

    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)

        lblName.text = binding!!.name
        lblAuthor.text = binding!!.author
        lblDescription.text = binding!!.description

    }

    companion object {

        private val ARG_BINDING = "ARG_BINDING"

        fun newInstance(binding: OHBinding): BindingFragment {
            val fragment = BindingFragment()
            val args = Bundle()
            args.putString(ARG_BINDING, GsonHelper.createGsonBuilder().toJson(binding))
            fragment.arguments = args
            return fragment
        }
    }
}// Required empty public constructor
